gcore/src/prog.rs

```diff
@@ -109,18 +115,30 @@ pub fn create_program_with_gas(
     gas_limit: u64,
     value: u128,
 ) -> ActorId {
+    let mut inner = [0u8; 56];
+    let fix_len_data = code_hash
+        .iter()
+        .copied()
+        .chain(gas_limit.to_le_bytes())
+        .chain(value.to_le_bytes());
+    populate_from_iterator(&mut inner, fix_len_data);
+
     unsafe {
         let mut program_id = ActorId::default();
         sys::gr_create_program_wgas(
-            code_hash.as_slice().as_ptr(),
+            inner.as_slice().as_ptr(),
             salt.as_ptr(),
             salt.len() as _,
             payload.as_ptr(),
             payload.len() as _,
-            gas_limit,
-            value.to_le_bytes().as_ptr(),
             program_id.as_mut_slice().as_mut_ptr(),
         );
         program_id
     }
 }
+
+fn populate_from_iterator(src: &mut [u8], data: impl Iterator<Item = u8>) {
```

Comment:
Have you reinvented [copy_from_slice()](https://doc.rust-lang.org/core/primitive.slice.html#method.copy_from_slice)?

Reply:
oh man...
I forgot I could populate any region of `inner`... thanks!